nomad/leader.go

```diff
@@ -289,6 +289,8 @@ func (s *Server) establishLeadership(stopCh chan struct{}) error {
 		return err
 	}
 
+	s.setConsistentReadReady()
```

Comment:
Is there a reason we need to wait till the end of this method? This method is called after the barrier write, so any reason not to toggle this at the beginning of this method rather than wait for all the other housekeeping that happens in this method?

Reply:
I was being defensive for fear of an RPC handling accessing/mutating brokers before they are enabled.  I'd hope that establishing leadership is a quick call and happens rarely such that optimizing to find exact ideal place to set consistent readiness isn't worth it.  What do you think?

Reply:
Yeah agreed that its a minor optimization, lets leave it as is